src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -923,6 +923,12 @@ private static void CheckRestrictedTypeReceiver(BoundExpression expression, Comp
                     // If this was a ref extension method, receiverArgument must be checked for L-value constraints.
                     // This helper method will also replace it with a BoundBadExpression if it was invalid.
                     receiverArgument = CheckValue(receiverArgument, BindValueKind.RefOrOut, diagnostics);
+
+                    CheckFeatureAvailability(receiverArgument.Syntax, MessageID.IDS_FeatureRefExtensionMethods, diagnostics);
```

Comment:
not sure if this is a big problem. It would require user to ignore the error at declaration and continue to making another one at the call site. Unless you are pasting a blob of code both declaring and using a ref extension - just does not seem to be common enough issue to have special case for that.

---
In reply to: [123672443](https://github.com/dotnet/roslyn/pull/20403#discussion_r123672443) [](ancestors = 123672443)